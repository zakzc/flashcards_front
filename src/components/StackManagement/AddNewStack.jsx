import React, { Component } from "react";
import { Redirect } from "react-router-dom";
// hooks
import useDB_Connection from "../../Data/DB-hook/connection-hook";
// Icons
import Consolidate from "../icons/consolidate";
import Plus from "../icons/plus";
import Right from "../icons/right";

export default class AddNewStack extends Component {
  constructor(props) {
    super(props);
    this.state = {
      currentUser: props.currentUser,
      currentStack: props.currentStack,
      messageToUser:
        "In order to make the new card, click on plus (+). After you finish adding at least 3 new cards, click on consolidate button.",
      messageToUserBack: "Your stack must have at least 3 cards.",
      frontValue: "",
      backValue: "",
      newFront: "",
      newBack: "",
      newStackName: "",
      newCardsToStack: [],
      readyToSend: true,
      numberOfCardsAdded: 0,
      redirect: false,
    };
    this.handleSubmitAddCardHandler = this.handleSubmitAddCardHandler.bind(
      this
    );
    this.handleFrontCardChange = this.handleFrontCardChange.bind(this);
    this.StackNameSubmitHandler = this.StackNameSubmitHandler.bind(this);
    this.AddCardForm = this.AddCardForm.bind(this);
    this.consolidateNewStack = this.consolidateNewStack.bind(this);
    this.handleSubmitReady = this.handleSubmitReady.bind(this);
    this.consolidateButtonStack = this.consolidateButtonStack.bind(this);
    this.infoCard = this.infoCard.bind(this);
    this.checkButton = this.checkButton.bind(this);
    this.AddCardButton = this.AddCardButton.bind(this);
    this.renderRedirect = this.renderRedirect.bind(this);
  }

  consolidateNewStack(newStack) {
    // let requestBody = newStack;
    let requestBody = JSON.stringify(newStack);
    useDB_Connection(
      "http://localhost:5000/cardApi/addNewStack",
      "POST",
      requestBody,
      {
        "Content-Type": "application/json",
        Accept: "application/json",
      }
    );
  }

  StackNameSubmitHandler = (event) => {
    event.preventDefault();
    let newName = this.stackNewName.value;
    console.log("--> ", newName);
    this.setState((state) => {
      return {
        newStackName: newName,
      };
    });
    console.log("stack name handler");
    console.table(this.state);
  };

  checkButton() {
    return (
      <div id="editCardButton_Positioning">
        <div className="flipContainer">
          <div className="flipInner">
            <button
              className="buttonStyle"
              type="button"
              onClick={this.StackNameSubmitHandler}
            >
              <div className="flipFront">
                <Right className="buttonStyle" />
              </div>
              <div className="flipBack">
                <span className="buttonMessage">Add name</span>
              </div>
            </button>
          </div>
        </div>
      </div>
    );
  }

  consolidateButtonStack() {
    if (this.state.numberOfCardsAdded > 2) {
      return (
        <div id="consolidateButton_Positioning">
          <div className="flipContainer">
            <div className="flipInner">
              <button
                className="buttonStyle"
                type="button"
                onClick={this.handleSubmitReady}
              >
                <div className="flipFront">
                  <Consolidate className="consolidateStackIcon" />
                </div>
                <div className="flipBack">
                  <span className="buttonMessage">CONSOLIDATE Changes</span>
                </div>
              </button>
            </div>
          </div>
        </div>
      );
    } else {
      return <div></div>;
    }
  }

  infoCard() {
    return (
      <div id="infoCardWrap">
        <div className="flipInfoCard">
          <div className="flipInfoCardInner">
            <p className="flipInfoCardFront">{this.state.messageToUser}</p>
            <p className="flipInfoCardBack">{this.state.messageToUserBack}</p>
          </div>
        </div>
      </div>
    );
  }

  handleSubmitReady() {
    this.setState({
      messageToUser:
        "Consolidating changes and sending the new data to your database.",
    });
    console.table(this.state);
    console.log(
      "Ready to submit",
      this.state.newStackName,
      this.state.currentUser.id,
      this.state.newFront,
      this.state.newBack
    );
    const newStack = {
      stackName: this.state.newStackName,
      createdBy: this.state.currentUser.id,
      cards: this.state.newCardsToStack,
    };
    console.log("NewStack", newStack);
    this.consolidateNewStack(newStack);
    this.setState({ redirect: true });
  }

  handleSubmitAddCardHandler = (event) => {
    event.preventDefault();
    this.setState({
      messageToUser:
        "Now add more cards and, when done, click on consolidate changes.",
    });
    this.state.newCardsToStack.push({
      front: this.state.newFront,
      back: this.state.newBack,
    });
    this.setState({
      frontValue: "",
      backValue: "",
      numberOfCardsAdded: this.state.numberOfCardsAdded + 1,
    });
    console.log("new cards to stack: ", this.state.newCardsToStack);
  };

  // View Methods

  renderRedirect() {
    if (this.state.redirect) {
      return (
        <Redirect
          to="/consolidateChanges"
          userIsLoggedIn={this.state.userIsLoggedIn}
        />
      );
    }
  }

  AddCardButton() {
    return (
      <div id="addCardButton_Positioning">
        <div className="flipContainer">
          <div className="flipInner">
            <button
              className="buttonStyle"
              type="button"
              onClick={this.handleSubmitAddCardHandler}
            >
              <div className="flipFront">
                <Plus className="addCardIcon" />
              </div>
              <div className="flipBack">
                <span className="buttonMessage">Add new card</span>
              </div>
            </button>
          </div>
        </div>
      </div>
    );
  }

  handleFrontCardChange = (e) => {
    this.setState({ frontValue: e.target.value });
    let addFront = e.target.value;
    console.log("handle: ", addFront);
    if (addFront) {
      this.setState({
        newFront: addFront,
        messageToUser: "don't forget to add the back of the card.",
      });
    }
  };

  handleBackCardChange = (e) => {
    this.setState({ backValue: e.target.value });
    let addBack = e.target.value;
    console.log("handle back: ", addBack);
    if (addBack) {
      this.setState({
        newBack: addBack,
        messageToUser: "Did you already add the back of the card?",
      });
    }
  };

  AddCardForm() {
    return (
      <React.Fragment>
        <div>
          <form>
            <div className="formElement">
              <label>Front of the card</label>
              <input
                className="cardInput"
                type="text"
                name="front"
                value={this.state.frontValue}
                onChange={this.handleFrontCardChange}
              />
            </div>
            <div className="formElement">
              <label>Back of the card</label>
              <input
                className="cardInput"
                type="text"
                name="front"
                value={this.state.backValue}
                onChange={this.handleBackCardChange}
              />
            </div>
            <br />
            <div id="AddCardButtons">
              <div>{this.AddCardButton()}</div>
              <div> {this.consolidateButtonStack()}</div>
            </div>
          </form>
          {this.infoCard()}
          {this.renderRedirect()}
        </div>
      </React.Fragment>
    );
  }

  newStackNameForm() {
    return (
      <React.Fragment>
        {this.state.newStackName >= 0 ? (
          <div id="letsAddSection">
            <h4>
              Let's <strong className="statInfo"> add </strong> a new stack,
              shall we? First, what is the name of your new stack:
            </h4>
            <form>
              <input
                id="formAddNewStack"
                type="text"
                ref={(input) => (this.stackNewName = input)}
              ></input>
              <div>{this.checkButton()}</div>
            </form>
          </div>
        ) : (
          <div id="addCardsToStack">
            <p>
              Great, so, the name is: <strong>{this.state.newStackName}</strong>
            </p>
            <p>
              Now, let's add <strong>one</strong> card to it. After that, go to{" "}
              <strong>Manage Your Stacks</strong> to keep adding more cards to
              it.
            </p>
            {this.AddCardForm()}
          </div>
        )}
      </React.Fragment>
    );
  }

  render() {
    return <React.Fragment>{this.newStackNameForm()}</React.Fragment>;
  }
}
