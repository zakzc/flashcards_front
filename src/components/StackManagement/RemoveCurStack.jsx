import React, { Component } from "react";
import { Redirect } from "react-router-dom";
// Data update
import RemoveCurrentStack from "../../Data/Data_Update/removeCurStack";
// Icon
import Trash from "../icons/trash";
// In case there are no stacks
import sampleStack from "../../Data/DummyData/sampleCards.json";

export default class RemoveCurStack extends Component {
  constructor(props) {
    super(props);
    this.state = {
      currentUser: props.currentUser,
      currentStack: props.currentStack,
      stackName: props.stackName,
      token: props.token,
      redirectUser: false,
      messageToUser:
        "Clicking the button will delete the stack. If you want to delete another stack, you must go to current stack and select the stack you want to delete.",
    };
    // DB actions
    // this.removeThisStack = RemoveCurrentStack;
    // methods from app.js
    this.setCurrentStack = props.setCurrentStack;
    // local
    this.warning = this.warning.bind(this);
    this.deleteButton = this.deleteButton.bind(this);
    this.deleteHandler = this.deleteHandler.bind(this);
    this.deleteStackView = this.deleteStackView.bind(this);
    // fallback stack
    this.sampleStack = sampleStack;
  }
  ////////////////
  // Implementation methods
  ////////////////

  async deleteHandler() {
    console.log(
      "Check pre deletion: ",
      this.state.currentStack._id,
      this.state.currentUser.userStacks.length
    );
    if (this.state.currentUser.userStacks.length === 1) {
      this.setState({
        messageToUser:
          "You cannot delete your only stack. You need to have at least one stack.",
      });
    } else {
      let removeProcess;
      removeProcess = await RemoveCurrentStack(
        this.state.currentStack._id,
        this.state.token
      )
        .then(() => {
          if (
            this.state.currentUser.userStacks[0].stack_id &&
            this.state.currentUser.userStacks[0].stack_id !== undefined
          ) {
            this.setCurrentStack(this.state.currentUser.userStacks[0].stack_id);
          } else {
            this.setCurrentStack(sampleStack);
          }
        })
        .then(() => {
          this.setState({ redirectUser: true });
        })
        .catch((err) => {
          console.log("Error on deletion process (error 36). ", err);
        });
      if (!removeProcess) {
        removeProcess = false;
        return removeProcess;
      }
    }
  }

  ////////////////
  // View methods
  ////////////////

  deleteButton() {
    return (
      <button
        // className="buttonStyle"
        className="buttonStyle"
        type="button"
        onClick={this.deleteHandler}
      >
        <div className="flipContainer">
          <div className="flipInner">
            <div className="flipFront">
              <Trash />
            </div>
            <div className="flipBack">
              <span className="buttonMessage">Confirm DELETE</span>
            </div>{" "}
          </div>
        </div>
      </button>
    );
  }

  warning() {
    return (
      <div>
        <h3>
          Your current stack is:
          <strong className="currentSelectedStack">
            {" "}
            {this.state.stackName}
          </strong>
        </h3>
        <div className="infoCard">{this.state.messageToUser}</div>
        <p></p>
        <p>
          But, before you continue,are you sure about this? This action can't be
          undone.
        </p>
      </div>
    );
  }

  redirectToConsolidate() {
    return (
      <div>
        <Redirect to="/consolidateChanges" />
      </div>
    );
  }

  deleteStackView() {
    return (
      <div id="deleteCurStackLayout">
        <div id="deleteWarning">{this.warning()}</div>
        <div id="deleteButtonConfirmation">{this.deleteButton()}</div>
        {this.state.redirectUser ? this.redirectToConsolidate() : <div></div>}
      </div>
    );
  }

  render() {
    return <div id="deleteStackView"> {this.deleteStackView()}</div>;
  }
}
