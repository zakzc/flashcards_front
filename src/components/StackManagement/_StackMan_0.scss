//////////////
// * General Layout
//////////////

#chosenFunctionalityArea {
  grid-area: "functionality";
}

#stackNavLeft {
  grid-area: stackNavLeft;
}

#stackNavCenter {
  grid-area: stackNavCenter;
}
#stackNavRight {
  grid-area: stackNavRight;
}

#StackNav {
  grid-area: "stackNav";
  display: grid;
  // grid-template-columns: 7fr, 1fr;
  grid-template-areas: ". stackNavLeft . stackNavCenter . stackNavRight .";
  // margin-top: 2%;
  // margin-bottom: 5%;
}

#stackMessage {
  grid-area: stackMessage;
}

#stackManLayout {
  display: grid;
  grid-auto-flow: row;
  grid-template-areas:
    "stackMessage"
    "stackNav"
    "functionality"
    ".";
  gap: 25%;
  grid-template-rows: 1fr, 1fr, 2fr 1fr;
  background-color: $mainBackColor;
  // adjust of the container positioning
  margin-top: 1%;
  margin-left: 1%;
  padding-bottom: 20%;
}

//////////////////////
// * Stack Nav buttons
//////////////////////

.buttonIsActive {
  color: $mainBackColor;
  position: absolute;
  background-color: transparent;
  // text-decoration: none;
  border: none;
  margin: none;
  padding: none;
  border-radius: 20px;
}

.buttonIsOff {
  color: $mainBackColor;
  position: absolute;
  background-color: transparent;
  // text-decoration: none;
  border: none;
  margin: none;
  padding: none;
  border-radius: 20px;
}

.buttonIsActive .navIconStackMan {
  background: transparent;
  color: $backColorNav;
  border-radius: 20px;
}

.buttonIsOff .navIconStackMan {
  background: transparent;
  color: $contrastColor;
}

#AddFront {
  background-color: $backColorNav;
  border-radius: 20px;
  display: block;
}

// Flip effect on Navigation buttons:
#navButtonContainer {
  background-color: $mainBackColor;
  border: none;
  text-decoration: none;
  border-radius: 20px;
}
