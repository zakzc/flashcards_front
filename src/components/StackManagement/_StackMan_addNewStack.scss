.cardInput {
  margin-left: 2%;
}

#addCardsToStack {
  margin-top: -8%;
}

#consolidateButton_Positioning {
  float: left;
  margin-left: 1%;
}

.addCardIcon {
  color: $contrastColor;
  background-color: $backColorNav;
}

.consolidateStackIcon {
  color: $contrastColor;
  background-color: $backColorNav;
}

#addCardWrapper {
  float: left;
  border-radius: 20px;
  background-color: $contrastColor;
  position: center;
  text-align: center;
  vertical-align: middle;
}

#consolidateWrapper {
  border-radius: 20px;
  float: right;
  background-color: $contrastColor;
  // width: 120px;
  // height: 100px;
  position: center;
  text-align: center;
  vertical-align: middle;
}

#addCardButton {
  background-color: $contrastColor;
  border-radius: 20px;
}

