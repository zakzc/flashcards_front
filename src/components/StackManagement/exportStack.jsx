import React, { Component } from "react";
// Icons
import Download from "../icons/export";
// hooks
// import useDB_Connection from "../../Data/DB-hook/connection-hook";
// Icons

export default class ExportStack extends Component {
  constructor(props) {
    super(props);
    this.state = {
      currentUser: props.currentUser,
      currentStack: props.currentStack,
    };
    // methods
    this.prepareData = this.prepareData.bind(this);
    this.exportHandler = this.exportHandler.bind(this);
    this.exportButton = this.exportButton.bind(this);
    this.infoOnExport = this.infoOnExport.bind(this);
  }

  ////////////////
  // Implemntation methods
  ////////////////

  prepareData() {
    // Data selection: omits the '_id' and 'created by id' of the stack for security reasons
    let dataToExport = this.state.currentStack;
    delete dataToExport.id;
    dataToExport.createdBy =
      this.state.currentUser.firstName + " " + this.state.currentUser.lastName;
    // Data preparation
    let data = JSON.stringify(dataToExport, null, 4);
    console.log("export:", data, typeof data);
    this.exportHandler(data);
  }

  exportHandler(stackData) {
    // Data exporting to file
    const a = document.createElement("a");
    a.href = URL.createObjectURL(new Blob([stackData], { type: "text/plain" }));
    a.setAttribute("download", "yourStackData.txt");
    document.body.appendChild(a);
    a.click();
    document.body.removeChild(a);
    console.log("Export handler");
  }

  ////////////////
  // View methods
  ////////////////

  exportButton() {
    return (
      <div id="exportStackButton_Positioning">
        <div className="flipContainer">
          <div className="flipInner">
            <button
              className="buttonStyle"
              type="button"
              onClick={this.prepareData}
            >
              <div className="flipFront">
                <Download />
              </div>
              <div className="flipBack">
                <span className="buttonMessage">Download</span>
              </div>
            </button>
          </div>
        </div>
      </div>
    );
  }

  infoOnExport() {
    return (
      <React.Fragment>
        <div>
          <h3>
            Your current stack is:
            <strong className="currentSelectedStack">
              {" "}
              {this.state.currentStack.stackName}
            </strong>
          </h3>
          <p>
            Clicking the button will export the current stack to a file. This
            file will be downloaded by your browser. Depending on local
            system/OS configurations, you night need to authorize this or allow
            it.
          </p>
        </div>
      </React.Fragment>
    );
  }

  render() {
    return (
      <React.Fragment>
        <div id="exportCurStackWrapper">
          <div>{this.infoOnExport()}</div>
          <div>{this.exportButton()}</div>
        </div>
      </React.Fragment>
    );
  }
}
