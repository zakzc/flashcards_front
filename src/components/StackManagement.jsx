import React, { Component } from "react";
import { Redirect } from "react-router-dom";
// Data Methods
import RemoveCurStack from "./StackManagement/RemoveCurStack";
import AddNewStack from "./StackManagement/AddNewStack";
import ExportStack from "./StackManagement/exportStack";
// DB
// import { useDB_Connection } from "../DB/DB-hook/connection-hook";
// Standard Log out page
import UserIsLoggedOut from "./views/UserIsLoggedOut";
// Icon
import Plus from "./icons/plus";
import Trash from "./icons/trash";
import Export from "./icons/export";

export default class StackManagement extends Component {
  constructor(props) {
    super(props);
    this.state = {
      userIsLoggedIn: props.userIsLoggedIn,
      currentUser: props.currentUser,
      currentStack: props.currentStack,
      add: true,
      export: false,
      delete: false,
    };
    // * Methods
    // hooks
    // this.useDB_Connection = useDB_Connection;
    // methods from app.js
    this.setCurrentStack = props.setCurrentStack;
    // local
    this.UserIsLoggedOut = UserIsLoggedOut;
    this.AddSwitch = this.AddSwitch.bind(this);
    this.exportSwitch = this.exportSwitch.bind(this);
    this.deleteSwitch = this.deleteSwitch.bind(this);
    this.addStackButton = this.addStackButton.bind(this);
    this.exportStackButton = this.exportStackButton.bind(this);
    this.deleteStackButton = this.deleteStackButton.bind(this);
    this.stackNav = this.stackNav.bind(this);
    this.chosenFunctionality = this.chosenFunctionality.bind(this);
    this.StackManagementLayout = this.StackManagementLayout.bind(this);
  }

  ////////////////
  // Implementation methods
  ////////////////

  AddSwitch() {
    this.setState((state) => ({
      add: true,
      export: false,
      delete: false,
    }));
  }

  exportSwitch() {
    this.setState((state) => ({
      add: false,
      export: true,
      delete: false,
    }));
  }

  deleteSwitch() {
    this.setState((state) => ({
      add: false,
      export: false,
      delete: true,
    }));
  }

  ////////////////
  // View methods
  ////////////////

  addStackButton() {
    return (
      <div id="navButton">
        <div id="addStackButton_Positioning">
          <div className="flipContainer">
            <div className="flipInner">
              <button
                // className="buttonStyle"
                className={this.state.add ? "buttonIsActive" : "buttonIsOff"}
                type="button"
                onClick={this.AddSwitch}
              >
                <div className="flipFront">
                  <Plus className="navIconStackMan" />
                </div>
                <div className="flipBack">
                  <span className="buttonMessage">NEW stack</span>
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
    );
  }

  exportStackButton() {
    return (
      <div id="navButton">
        <div id="exportStackButton_Positioning">
          <div className="flipContainer">
            <div className="flipInner">
              <button
                // className="buttonStyle"
                className={this.state.export ? "buttonIsActive" : "buttonIsOff"}
                type="button"
                onClick={this.exportSwitch}
              >
                <div className="flipFront">
                  <Export className="navIconStackMan" />
                </div>
                <div className="flipBack">
                  <span className="buttonMessage">DOWNLOAD Stack</span>
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
    );
  }

  deleteStackButton() {
    return (
      <div id="navButton">
        <div id="deleteStackButton_Positioning">
          <div className="flipContainer">
            <div className="flipInner">
              <button
                // className="buttonStyle"
                className={this.state.delete ? "buttonIsActive" : "buttonIsOff"}
                type="button"
                onClick={this.deleteSwitch}
              >
                <div className="flipFront">
                  <Trash className="navIconStackMan" />
                </div>
                <div className="flipBack">
                  <span className="buttonMessage">DELETE stack</span>
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
    );
  }

  stackNav() {
    return (
      <div className="StackNavAdjust">
        <div id="stackNavButtons">
          <div id="stackNavLeft" className="optionIsActive">
            {this.addStackButton()}
          </div>
          <div id="stackNavCenter">{this.exportStackButton()}</div>
          <div id="stackNavRight">{this.deleteStackButton()}</div>
        </div>
      </div>
    );
  }

  chosenFunctionality() {
    return (
      <div id="chosenFunctionalityArea">
        {this.state.add ? (
          <AddNewStack
            userIsLoggedIn={this.state.userIsLoggedIn}
            currentUser={this.state.currentUser}
            currentStack={this.state.currentStack}
          />
        ) : (
          <div></div>
        )}
        {this.state.export ? (
          <ExportStack
            userIsLoggedIn={this.state.userIsLoggedIn}
            currentUser={this.state.currentUser}
            currentStack={this.state.currentStack}
            // setCurrentStack={this.setCurrentStack}
          />
        ) : (
          <div></div>
        )}
        {this.state.delete ? (
          <RemoveCurStack
            userIsLoggedIn={this.state.userIsLoggedIn}
            currentUser={this.state.currentUser}
            currentStack={this.state.currentStack}
            setCurrentStack={this.setCurrentStack}
          />
        ) : (
          <div></div>
        )}
      </div>
    );
  }

  StackManagementLayout() {
    return (
      <React.Fragment>
        <div id="stackManWrapper">
          {this.stackNav()}
          {this.chosenFunctionality()}
        </div>
      </React.Fragment>
    );
  }

  StackManagementView() {
    console.log("STACK MANAGEMENT");
    return (
      <React.Fragment>
        <div id="cardManagementView">
          {this.state.userIsLoggedIn ? (
            <div id="loggedInView">{this.StackManagementLayout()}</div>
          ) : (
            <Redirect to="/login" />
          )}
        </div>
      </React.Fragment>
    );
  }

  render() {
    return (
      <React.Fragment>
        <div id="StackManagementView">{this.StackManagementView()}</div>
      </React.Fragment>
    );
  }
}
