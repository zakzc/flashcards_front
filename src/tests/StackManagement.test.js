import React from "react";
import Enzyme from "enzyme";
///
import Adapter from "enzyme-adapter-react-16";
import { shallow } from "enzyme";
///
// Parent
import StackManagement from "../components/StackManagement/StackMan_0";
// Children
import AddNewStack from "../components/StackManagement/StackMan_AddNewStack";
import ExportNewStack from "../components/StackManagement/StackMan_ExpoStack";
import RemoveCurrentStack from "../components/StackManagement/StackMan_RemoveCurStack";

Enzyme.configure({ adapter: new Adapter() });

// Parent
describe("\n __| Stack Management |__", () => {
  const stackManagement = shallow(<StackManagement />);

  describe("Initialization process", () => {
    it("Renders the CardSet component", () => {
      expect(stackManagement).toMatchSnapshot();
    });

    it("Initializes the State", () => {
      expect(stackManagement.state().add).toEqual(true);
      expect(stackManagement.state().export).toEqual(false);
      expect(stackManagement.state().delete).toEqual(false);
    });

    describe("Imports", () => {
      test("RemoveCurStack", () => {
        expect(stackManagement.find("RemoveCurStack")).toBeTruthy();
      });
      test("AddNewStack", () => {
        expect(stackManagement.find("AddNewStack")).toBeTruthy();
      });
      test("ExportStack", () => {
        expect(stackManagement.find("ExportStack")).toBeTruthy();
      });
      test("UserIsLoggedOut", () => {
        expect(stackManagement.find("UserIsLoggedOut")).toBeTruthy();
      });
      describe("Svg imports", () => {
        test("Plus", () => {
          expect(stackManagement.find("Plus")).toBeTruthy();
        });
        test("Trash", () => {
          expect(stackManagement.find("Trash")).toBeTruthy();
        });
        test("Export", () => {
          expect(stackManagement.find("Export")).toBeTruthy();
        });
      });
    });
  });
});

// Children
describe("\n ___| Add New Stack |___", () => {
  const addNewStack = shallow(<AddNewStack />);

  describe("\tInitialization process", () => {
    it("Renders the Stack Management component", () => {
      expect(addNewStack).toMatchSnapshot();
    });

    it("Initializes the State", () => {
      expect(addNewStack.state().frontValue).toEqual("");
      expect(addNewStack.state().backValue).toEqual("");
      expect(addNewStack.state().newFront).toEqual("");
      expect(addNewStack.state().newBack).toEqual("");
      expect(addNewStack.state().newStackName).toEqual("");
      expect(addNewStack.state().newCardsToStack).toEqual([]);
      expect(addNewStack.state().readyToSend).toEqual(true);
      expect(addNewStack.state().numberOfCardsAdded).toEqual(0);
      expect(addNewStack.state().redirect).toEqual(false);
    });

    describe("Imports", () => {
      test("useDB_Connection", () => {
        expect(addNewStack.find("useDB_Connection")).toBeTruthy();
      });
      describe("Svg imports", () => {
        test("Consolidate", () => {
          expect(addNewStack.find("Consolidate")).toBeTruthy();
        });
        describe("Svg imports", () => {
          test("Plus", () => {
            expect(addNewStack.find("Plus")).toBeTruthy();
          });
        });
        describe("Svg imports", () => {
          test("Right", () => {
            expect(addNewStack.find("Right")).toBeTruthy();
          });
        });
      });
    });
  });
});

describe("\n ___| Export New Stack |___", () => {
  const exportNewStack = shallow(<ExportNewStack />);

  describe("\tInitialization process", () => {
    it("Renders the ManageCards component", () => {
      expect(exportNewStack).toMatchSnapshot();
    });

    describe("Imports", () => {
      describe("Svg imports", () => {
        test("Download", () => {
          expect(exportNewStack.find("Download")).toBeTruthy();
        });
      });
    });
  });
});

describe("\n ___| Remove current stack |___ ", () => {
  const removeCurrentStack = shallow(<RemoveCurrentStack />);

  describe("\tInitialization process", () => {
    it("Renders the ManageCards component", () => {
      expect(removeCurrentStack).toMatchSnapshot();
    });

    it("Initializes the State", () => {
      expect(removeCurrentStack.state().redirectUser).toEqual(false);
    });
  });

  describe("\tImports", () => {
    test("updateCards", () => {
      expect(removeCurrentStack.find("RemoveCurrentStack")).toBeTruthy();
    });
    describe("\tSvg imports", () => {
      test("Trash", () => {
        expect(removeCurrentStack.find("Trash")).toBeTruthy();
      });
    });
  });
});
