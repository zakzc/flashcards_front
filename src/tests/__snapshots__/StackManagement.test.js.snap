// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`
 ___| Add New Stack |___ 	Initialization process Renders the Stack Management component 1`] = `ShallowWrapper {}`;

exports[`
 ___| Export New Stack |___ 	Initialization process Renders the ManageCards component 1`] = `ShallowWrapper {}`;

exports[`
 ___| Remove current stack |___  	Initialization process Renders the ManageCards component 1`] = `ShallowWrapper {}`;

exports[`
 __| Stack Management |__ Initialization process Renders the CardSet component 1`] = `ShallowWrapper {}`;
